src/component.py

```diff
@@ -2,162 +2,255 @@
             cols = [c[0] for c in table_meta]
 
             tm = TableMetadata()
             tm.add_column_data_types({c[0]: self.convert_base_types(c[1]) for c in table_meta})
-            out_table = self.create_out_table_definition(f"{o}.csv", columns=cols, table_metadata=tm)
+            out_table = self.create_out_table_definition(f"{table_name}.csv", columns=cols, table_metadata=tm,
+                                                         primary_key=primary_key, incremental=incremental,
+                                                         destination=destination)
             self.write_manifest(out_table)
 
-            conn.execute(f"COPY '{o}' TO '{out_table.full_path}' (HEADER, DELIMITER ',')")
+            self._connection.execute(f'''COPY "{table_name}" TO "{out_table.full_path}"
+                                        (HEADER, DELIMITER ',', FORCE_QUOTE *)''')
 
-        conn.close()
+        self._connection.close()
         self.move_files()
 
-    def run_simple_query(self, input_table: TableDefinition, query: str):
-        table_name = input_table.name.replace(".csv", "")
+    def run_simple_query(self, q: dict):
+        self.create_table(q["input"])
 
-        with self.db_connection() as conn:
-            self.create_table(conn, input_table)
-            logging.info(f"Table {table_name} created.")
+        incremental = False
+        primary_key = []
 
-            conn.execute(f"CREATE OR REPLACE TABLE '{table_name}' AS {query};")
-            logging.info(f"Table {table_name} query finished.")
+        if isinstance(q.get("output"), dict):
+            incremental = q["output"].get('incremental', False)
+            primary_key = q["output"].get('primary_key', [])
 
-            table_meta = conn.execute(f"""DESCRIBE TABLE '{table_name}';""").fetchall()
-            cols = [c[0] for c in table_meta]
+        table_meta = self._connection.execute(f"""DESCRIBE {q["query"]};""").fetchall()
+        cols = [c[0] for c in table_meta]
 
-            tm = TableMetadata()
-            tm.add_column_data_types({c[0]: self.convert_base_types(c[1]) for c in table_meta})
-            out_table = self.create_out_table_definition(f"{table_name}.csv", columns=cols, table_metadata=tm)
-            self.write_manifest(out_table)
+        tm = TableMetadata()
+        tm.add_column_data_types({c[0]: self.convert_base_types(c[1]) for c in table_meta})
 
-            conn.execute(f"COPY '{table_name}' TO '{out_table.full_path}' (HEADER, DELIMITER ',')")
-            logging.info(f"Table {table_name} export finished.")
+        if isinstance(q.get("input"), dict):
+            destination = q.get("input", {}).get('duckdb_destination') or q.get("input", {}).get("input_pattern")
+        else:
+            destination = q.get("input")
 
-    def create_table(self, conn: DuckDBPyConnection, table: TableDefinition) -> None:
+        if isinstance(q.get("output"), dict):
```

Comment:
Jenom kosmeticka ten q.get('output') je zbytecne zduplikovany. A co kdyz ten parametr prejmenujem? Obecne je lepsi to mit explicitne definovany v configuration. To je pro priste.